### PR TITLE
gcp - update getting started workflow

### LINKED ISSUE #5943: gcp - Getting Started Documentation out of date
**Describe the bug**
* `gcloud auth` command line syntax has since changed - login name is no longer supported
* Google Cloud Project is never explicitly (or implicitly) set - documenting this in the workflow will reduce friction

**To Reproduce**
1. Follow https://cloudcustodian.io/docs/gcp/gettingstarted.html
2. At `gcloud` command step, observe the following message:
   ```
   $ gcloud auth application-default login <...>@gmail.com
   ERROR: (gcloud.auth.application-default.login) unrecognized arguments: <...>@gmail.com 

   To search the help text of gcloud commands, run:
     gcloud help -- SEARCH_TERMS
   ```
3. After creating example policy and executing `custodian run --output-dir=. custodian.yaml `
   ```
   2020-07-11 10:12:53,849: google.auth._default:WARNING No project ID could be determined. Consider running `gcloud config set project` or setting the GOOGLE_CLOUD_PROJECT environment variable
   2020-07-11 10:12:54,526: custodian.output:ERROR Error while executing policy
   Traceback (most recent call last):
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n/policy.py", line 291, in run
       resources = self.policy.resource_manager.resources()
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 203, in resources
       resources = self._fetch_resources(q)
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 223, in _fetch_resources
       return self.augment(self.source.get_resources(query)) or []
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 86, in get_resources
       return self.query.filter(self.manager, **query)
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 45, in filter
       project = session.get_default_project()
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/client.py", line 195, in get_default_project
       raise ValueError("No GCP Project ID set - set CLOUDSDK_CORE_PROJECT")
   ValueError: No GCP Project ID set - set CLOUDSDK_CORE_PROJECT
   2020-07-11 10:12:54,527: custodian.commands:ERROR Error while executing policy my-first-policy, continuing
   Traceback (most recent call last):
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n/commands.py", line 282, in run
       policy()
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n/policy.py", line 1176, in __call__
       resources = mode.run()
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n/policy.py", line 291, in run
       resources = self.policy.resource_manager.resources()
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 203, in resources
       resources = self._fetch_resources(q)
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 223, in _fetch_resources
       return self.augment(self.source.get_resources(query)) or []
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 86, in get_resources
       return self.query.filter(self.manager, **query)
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/query.py", line 45, in filter
       project = session.get_default_project()
     File "/home/marco/Projects/silph.co/silph-infra/c7n/.venv/lib/python3.8/site-packages/c7n_gcp/client.py", line 195, in get_default_project
       raise ValueError("No GCP Project ID set - set CLOUDSDK_CORE_PROJECT")
   ValueError: No GCP Project ID set - set CLOUDSDK_CORE_PROJECT
   2020-07-11 10:12:54,528: custodian.commands:ERROR The following policies had errors while executing
    - my-first-policy
   ```
   Suspiciously, running `gcloud config set project <...>` does not resolve the error despite what the output would suggest. Instead I had to set an environment variable of `GOOGLE_CLOUD_PROJECT=<...>`

**Expected behavior**
My expectations is - as best as possible - a papercut free getting started experience. Where by the end of the guide the examples executed completed without having to leave the getting started guide.

**Background (please complete the following information):**
 - OS: Ubuntu 20.04 LTS 5.4.0-33-generic #37-Ubuntu
 - Python Version: 3.8.2
 - Custodian Version: 0.9.4
 - Cloud Provider: gcp
 - Policy:
   ```yaml
   policies:
   - name: my-first-policy
     description: |
       Stops all compute instances that are named "test"
     resource: gcp.instance
     filters:
     - type: value
       key: name
       value: juju-278695-11
     actions:
     - type: stop
   ```
 - `custodian version --debug` output
   ```
   Custodian:   0.9.4
   Python:      3.8.2 (default, Apr 27 2020, 15:53:34) 
                [GCC 9.3.0]
   Platform:    posix.uname_result(sysname='Linux', nodename='T480', release='5.4.0-33-generic', version='#37-Ubuntu SMP Thu May 21 12:53:59 UTC 2020', machine='x86_64')
   Using venv:  True
   Docker: False
   Installed: 

   argcomplete==1.11.1
   attrs==19.3.0
   boto3==1.14.20
   botocore==1.17.20
   c7n==0.9.4
   cachetools==4.1.1
   contextlib2==0.6.0
   docutils==0.15.2
   google-api-core==1.21.0
   google-api-python-client==1.9.3
   google-auth==1.18.0
   google-auth-httplib2==0.0.4
   google-cloud-core==1.3.0
   google-cloud-logging==1.15.0
   google-cloud-monitoring==0.34.0
   google-cloud-storage==1.29.0
   google-resumable-media==0.5.1
   googleapis-common-protos==1.52.0
   httplib2==0.18.1
   importlib-metadata==1.7.0
   jmespath==0.10.0
   jsonschema==3.2.0
   protobuf==3.12.2
   pyasn1==0.4.8
   pyasn1-modules==0.2.8
   pyrsistent==0.16.0
   python-dateutil==2.8.1
   pytz==2020.1
   pyyaml==5.3.1
   ratelimiter==1.2.0.post0
   requests==2.22.0
   retrying==1.3.3
   rsa==4.6
   s3transfer==0.3.3
   setuptools==44.0.0
   six==1.15.0
   tabulate==0.8.7
   uritemplate==3.0.1
   urllib3==1.25.9
   zipp==3.1.0
```

### PR DESCRIPTION
**Pull request by marcoceppi**
fixes #5943

### CODE REVIEW

File: docs/source/gcp/gettingstarted.rst [hunk 1]
```diff
@@ -119,8 +118,9 @@ Next, run the following command to execute the policy with Custodian:
 
 .. code-block:: bash
 
-    custodian run --output-dir=. custodian.yml
+    GOOGLE_CLOUD_PROJECT="project-id" custodian run --output-dir=. custodian.yml
 
+Note: If you already have Google Cloud Project configured for gcloud or Cloud SDK, then you may omit providing it on the command line.
```
Comment (kapilt) [inline docs/source/gcp/gettingstarted.rst:123]: we don't actually source the gcloud config settings (config file), we just source the same env vars.

File: docs/source/gcp/gettingstarted.rst [hunk 2]
```diff
@@ -119,8 +118,9 @@ Next, run the following command to execute the policy with Custodian:
 
 .. code-block:: bash
 
-    custodian run --output-dir=. custodian.yml
+    GOOGLE_CLOUD_PROJECT="project-id" custodian run --output-dir=. custodian.yml
```
Comment (kapilt) [inline docs/source/gcp/gettingstarted.rst:121]: a random aside, we generally want try to use the env vars that the cli chooses, but google is internally schizophrenic and uses different env vars by different tools. The cli name squats on CLOUDSDK_CORE_PROJECT and the sdk uses several names though GOOGLE_CLOUD_PROJECT appears to be the latest naming flavor of the month.

File: docs/source/gcp/gettingstarted.rst (hunk 1 above)
Comment (marcoceppi) [reply docs/source/gcp/gettingstarted.rst:123]: I was wondering about this - I set the `gcloud` default project but it wasn't picked up. I instead mirrored what was in the AWS docs to suggest it's possible but I wasn't able to figure out

Comment (kapilt) [reply docs/source/gcp/gettingstarted.rst:123]: in aws identity creds identify target, and the sdks are aligned to the cli, such that the sdks automatically respect cli profile configurations. google instead produces software according to conway's law. the cli team doesn't use github, invented its own api client thats different then the published sdks (of which there are three for python) and doesn't respect sdk configuration, instead does its own thing idiosyncratic thing, which the sdk team then ignores. dysfunctional synergy google style.

nutshell we shouldn't document this as using the gcloud config, we should explicitly recommend setting the environment variable. as an application, we do support all the various environment variables (historical sdk, current sdk, cli) to try and provide some sanity to a user.

Comment (marcoceppi) [reply docs/source/gcp/gettingstarted.rst]: Sounds good, I've updated the PR to remove this note and continue using the currently suggested CLI/SDK environment variable

**Pull request merged**